src/LogFileReader.cpp

```diff
@@ -47,12 +47,25 @@ bool LogFileReader::readAll(const QString& fileName)
 // File input is a csv file in the format of hh:mm:ss:zzz, voltage, current
 bool LogFileReader::parseLine(const QString& line, BatteryData& batteryData) const
 {
-    // TODO implement this first
+    //example input= 07:01:07.024, 3.538654, 25.523571
+    bool convertedOkay;
+    QStringList lineSplit = line.split(",");
+    if(lineSplit.length()!= 3)
+        {return false;}
+
+    QTime time = QTime::fromString(lineSplit[0],"hh:mm:ss.zzz");
```

Comment:
You should move this `"hh.:mm:ss.zzz"` and `","` constant into an anonymous namespace constant variable at the top.